pydantic/errors.py

```diff
@@ -234,4 +234,9 @@ def __init__(self, *, expected_arbitrary_type) -> None:
 
 
 class JsonError(PydanticValueError):
-    msg_template = "Not valid JSON provided - it can't be decoded"
+    msg_template = "Invalid JSON"
+
+
+class JsonNotStrError(PydanticTypeError):
+    code = 'json_not_str'
```

Comment:
name `JsonTypeError` and change code to just `'json'` which will result in the final code being `type_error.json`